### ISSUE TITLE
Disable spell checker in link and xref descriptions

### ISSUE THREAD

**Issue by mmuehlfeldRH** [redhat-documentation/vale-at-red-hat#499]
**Is your feature request related to a problem? Describe.**
If you use the names of utilities or applications in xref and link descriptions, vale reports a warning. I have often several of these false positives in my vale reports.

Example:

`Configure the WireGuard server by using xref:proc_configuring-a-wireguard-by-using-nmcli_assembly_setting-up-a-wireguard-vpn[nmcli].`

causes:

```
29:86   warning     Use correct American English    RedHat.Spelling         
                     spelling. Did you really mean                           
                     'nmcli'? 
```    

**Describe the solution you'd propose**
Disable spell checker in link and xref descriptions.

**Describe alternatives you've considered**
Provide a simple way for writers to add the names of utilities, etc. to the spell checker list of vale.

**Comment by themr0c** [comment]
If `nmcli` is a command name, shouldn't it be in backticks?

This would not raise an alert:

```
Configure the WireGuard server by using xref:proc_configuring-a-wireguard-by-using-nmcli_assembly_setting-up-a-wireguard-vpn[`nmcli`]
```

**Comment by mmuehlfeldRH** [comment]
In normal text yes, but we don't use markups in titles or link descriptions because the rendering looks strange.

I found an example on the Customer Portal where markups are used in an xref:
![example](https://user-images.githubusercontent.com/43061675/228518697-f0f435c4-0f02-428a-a109-d23416fa949d.png)
It doesn't look like one single link with the markups.

**Comment by jafiala** [comment]
I agree. According to IBM Style, "In the label, try to include text that mirrors the title of the linked site". Because we cannot control titles of linked sites, we're bound to produce errors against the style which Vale would incorrectly flag.

Excluding the content of link and xref descriptions would produce fewer false positives and help writers see the relevant recommendations.

**Comment by aireilly** [comment]
I think we can fix this with scoping or similiar. need to investigate.

**Comment by aireilly** [comment]
eg., https://github.com/redhat-documentation/vale-at-red-hat/pull/452
